netbox_agent/lshw.py

```diff
@@ -131,6 +118,20 @@ def find_storage(self, obj):
                     self.disks.append(d)
             except Exception:
                 pass
+        elif "children" in obj:
+            for device in obj["children"]:
+                if device.get("class") == "disk":
+                    self.disks.append(
+                        {
+                            "logicalname": device.get("logicalname"),
+                            "product": device.get("product"),
+                            "serial": device.get("serial"),
+                            "version": device.get("version"),
+                            "size": device.get("size"),
+                            "description": device.get("description"),
+                            "type": device.get("description"),
+                        }
+                    )
```

Comment:
It seems some controllers are nested one level deeper, it could help for some edge cases to do a deep search inside all `storage` objects, at the end of `find_storage()`

```suggestion
        elif device.get("class") == "disk":
            self.disks.append(
                {
                    "logicalname": device.get("logicalname"),
                    "product": device.get("product"),
                    "serial": device.get("serial"),
                    "version": device.get("version"),
                    "size": device.get("size"),
                    "description": device.get("description"),
                    "type": device.get("description"),
                }
            )
        else:
            for child in obj.get("children") or []:
                self.find_storage(child)
```